File: backend/app/schemas/stock.py
"""股票搜索相关 Pydantic 模型"""

from pydantic import BaseModel


class StockItem(BaseModel):
    ts_code: str
    symbol: str
    name: str
    market: str


class StockSearchResponse(BaseModel):
    items: list[StockItem]
    total: int

File: frontend/src/types/stock.ts
export interface StockItem {
  ts_code: string;
  symbol: string;
  name: string;
  market: string;
}

export interface StockSearchResponse {
  items: StockItem[];
  total: number;
}

export interface KLineItem {
  trade_date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  pre_close: number | null;
  vol: number;
  amount: number;
}

export interface KLineData {
  ts_code: string;
  name: string;
  items: KLineItem[];
}

export interface ValuationData {
  ts_code: string;
  trade_date: string;
  pe_ttm: number | null;
  pe_static: number | null;
  pb: number | null;
  market_cap: number | null;
  circ_market_cap: number | null;
  dividend_yield: number | null;
}
